File: money/blackjack/python/card.py
from enum import *

class CardValue(IntEnum):
  ACE = 1
  TWO = 2
  THREE = 3
  FOUR = 4
  FIVE = 5
  SIX = 6
  SEVEN = 7
  EIGHT = 8
  NINE = 9
  TEN = 10
  JACK = 11
  QUEEN = 12
  KING = 13

class CardSuit(Enum):
  SPADES = 1
  DIAMONDS = 2
  CLUBS = 3
  HEARTS = 4

class Card:
  def __init__(self, value, suit):
    self.m_value = CardValue(value).value
    self.m_suit = CardSuit(suit)

  def __str__(self): #the print function
    tmpValue = ""
    tmpSuit = ""
    if self.m_value == CardValue.JACK:
      tmpValue = "J"
    elif self.m_value == CardValue.QUEEN:
      tmpValue = "Q"
    elif self.m_value == CardValue.KING:
      tmpValue = "K"
    elif self.m_value == CardValue.ACE:
      tmpValue = "A"
    else:
      tmpValue = str(self.m_value)

    if self.m_suit == CardSuit.SPADES:
      tmpSuit = "s"
    elif self.m_suit == CardSuit.DIAMONDS:
      tmpSuit = "d"
    elif self.m_suit == CardSuit.CLUBS:
      tmpSuit = "c"
    else: #elif self.m_suit = CardSuit.HEARTS:
      tmpSuit = "h"
    return tmpValue + tmpSuit

  def hard_value(self):
    if self.m_value == CardValue.JACK or \
       self.m_value == CardValue.QUEEN or \
       self.m_value == CardValue.KING:
      return 10
    else:
      return self.m_value

  def soft_value(self):
    if self.m_value == CardValue.JACK or \
       self.m_value == CardValue.QUEEN or \
       self.m_value == CardValue.KING:
      return 10
    elif self.m_value == CardValue.ACE:
      return 11
    else:
      return self.m_value

File: money/blackjack/c++/card.cpp
#include "card.h"

Card::Card(int value, int suit)
{
  switch(value)
  {
    case 0: m_value = ACE;    break;
    case 1: m_value = TWO;    break;
    case 2: m_value = THREE;  break;
    case 3: m_value = FOUR;   break;
    case 4: m_value = FIVE;   break;
    case 5: m_value = SIX;    break;
    case 6: m_value = SEVEN;  break;
    case 7: m_value = EIGHT;  break;
    case 8: m_value = NINE;   break;
    case 9: m_value = TEN;    break;
    case 10: m_value = JACK;  break;
    case 11: m_value = QUEEN; break;
    case 12: m_value = KING;  break;
    default: cout << "Bad Card Value." << endl;
  }
  switch(suit)
  {
    case 0: m_suit = SPADES;   break;
    case 1: m_suit = DIAMONDS; break;
    case 2: m_suit = CLUBS;    break;
    case 3: m_suit = HEARTS;   break;
    default: cout << "Bad Card Suit." << endl;
  }
}

string Card::print()
{
  string tmp;
  switch(m_value)
  {
    case ACE:   tmp = "A";  break;
    case TWO:   tmp = "2";  break;
    case THREE: tmp = "3";  break;
    case FOUR:  tmp = "4";  break;
    case FIVE:  tmp = "5";  break;
    case SIX:   tmp = "6";  break;
    case SEVEN: tmp = "7";  break;
    case EIGHT: tmp = "8";  break;
    case NINE:  tmp = "9";  break;
    case TEN:   tmp = "10"; break;
    case JACK:  tmp = "J";  break;
    case QUEEN: tmp = "Q";  break;
    case KING:  tmp = "K";  break;
  }
  switch(m_suit)
  {
    case SPADES:   tmp += "s"; break;
    case DIAMONDS: tmp += "d"; break;
    case CLUBS:    tmp += "c"; break;
    case HEARTS:   tmp += "h"; break;
  }
  return tmp;
}

string Card::print_blackjack()
{
  string tmp;
  switch(m_value)
  {
    case ACE:   tmp = "A";  break;
    case TWO:   tmp = "2";  break;
    case THREE: tmp = "3";  break;
    case FOUR:  tmp = "4";  break;
    case FIVE:  tmp = "5";  break;
    case SIX:   tmp = "6";  break;
    case SEVEN: tmp = "7";  break;
    case EIGHT: tmp = "8";  break;
    case NINE:  tmp = "9";  break;
    case TEN:   tmp = "10"; break;
    case JACK:  tmp = "J";  break;
    case QUEEN: tmp = "Q";  break;
    case KING:  tmp = "K";  break;
  }
  return tmp;
}

int Card::soft_blackjack_value()
{
  int tmp;
  switch(m_value)
  {
    case ACE:   tmp =  11;   break;
    case TWO:   tmp =  2;   break;
    case THREE: tmp =  3;   break;
    case FOUR:  tmp =  4;   break;
    case FIVE:  tmp =  5;   break;
    case SIX:   tmp =  6;   break;
    case SEVEN: tmp =  7;   break;
    case EIGHT: tmp =  8;   break;
    case NINE:  tmp =  9;   break;
    case TEN:   tmp =  10;  break;
    case JACK:  tmp =  10;  break;
    case QUEEN: tmp =  10;  break;
    case KING:  tmp =  10;  break;
  }
  return tmp;
}

int Card::hard_blackjack_value()
{
  int tmp;
  switch(m_value)
  {
    case ACE:   tmp =  1;   break;
    case TWO:   tmp =  2;   break;
    case THREE: tmp =  3;   break;
    case FOUR:  tmp =  4;   break;
    case FIVE:  tmp =  5;   break;
    case SIX:   tmp =  6;   break;
    case SEVEN: tmp =  7;   break;
    case EIGHT: tmp =  8;   break;
    case NINE:  tmp =  9;   break;
    case TEN:   tmp =  10;  break;
    case JACK:  tmp =  10;  break;
    case QUEEN: tmp =  10;  break;
    case KING:  tmp =  10;  break;
  }
  return tmp;
}
